site/en/blog/migrating-to-typescript/index.md

```diff
@@ -0,0 +1,166 @@
+
+In general, this approach worked with few issues.
+We ran into several bugs in the TypeScript compiler, but most of them were obscure:
+
+1. An optional parameter with a function type which returns `any` is treated as required: [#38551](https://github.com/microsoft/TypeScript/issues/38551)
+2. A property assignment to a static method of a class breaks declaration: [#38553](https://github.com/microsoft/TypeScript/issues/38553)
+3. The declaration of a subclass with a no-args constructor and a super-class with an args constructor omits the child constructor: [#41397](https://github.com/microsoft/TypeScript/issues/41397)
+
+These bugs highlight that, for the 99% case, the TypeScript compiler is a solid foundation to build upon.
+Yes, these obscure bugs would sometimes cause issues for DevTools, but most of the time they were obscure enough that we could easily work around them.
+
+The only issue that had caused some confusion was the non-deterministic output of `.tsbuildinfo` files: [#37156](https://github.com/microsoft/TypeScript/issues/37156).
+At Chromium, we require that any two builds of the same Chromium commit result in the exact same output.
+Unfortunately, our Chromium build engineers discovered that the `.tsbuildinfo` output was non-deterministic: [crbug.com/1054494](https://bugs.chromium.org/p/chromium/issues/detail?id=1054494).
+To workaround this issue, we had to monkey-patch the `.tsbuildinfo` file (which essentially contains JSON) and post-process it to return a deterministic output: [https://crrev.com/c/2091448](https://chromium-review.googlesource.com/c/devtools/devtools-frontend/+/2091448)
+Luckily, the TypeScript team resolved the upstream issue and we were soon able to remove our workaround. Thank you to the TypeScript team for being receptive to bug reports and fixing these issues promptly!
+
+Overall, we are pleased with the (type) correctness of the TypeScript compiler.
+We hope that Devtools as a large open-source JavaScript project has helped solidify JavaScript support in TypeScript.
+
+## Analyzing the aftermath
+
+We were able to make good progress in resolving these type errors and slowly increasing the amount of code checked by TypeScript.
+However, in August 2020 (9 months into this migration) we did a check-in and discovered we would not hit our deadline with our current pace.
+One of our engineers built an analysis graph to show the progress of "TypeScriptification" (the name we gave to this migration).
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/vxsdG9lI4YjuWi8ibZr9.png", alt="TypeScript Migration progress", width="800", height="639" %}
+*TypeScript Migration progress - Tracking lines of code remaining that need migrating*
```

Comment:
These asterisks remained in the eventual output. Do we need a newline before it?